src/Network/XMPP/Helpers.hs

```diff
@@ -46,7 +45,13 @@ connectViaTcp server port = do
 --  via HTTP 1.0 proxy
 connectViaHttpProxy :: Show a => HostName -> Integer -> T.Text -> a -> IO Handle
 connectViaHttpProxy proxyServer proxyPort server port = do
-  h <- connectTo proxyServer $ PortNumber $ fromIntegral proxyPort
```

Comment:
Both `connectTo` and `PortNumber` were removed in the newer version, so I am using `socket` in here. Let me know if you feel like it could break something

Reply:
Oh wow, you actually fixed this! That's great! I have so many issues to solved I just added a bound to cabal, but this is much better!